server/models/ParticipantsModel/index.js

```diff
@@ -42,26 +42,24 @@ const ParticipantsModel = {
         participant: {
           $in:
             groupedParticipants.length > 1
-              ? ParticipantsModel.intersectParticipants(
-                  groupedParticipants[0],
-                  groupedParticipants[1]
-                )
+              ? ParticipantsModel.intersectParticipants(groupedParticipants)
               : groupedParticipants.flat(),
         },
       })
       .stream()
   },
-  intersectParticipants: (listA, listB) => {
-    const smallestList = listA.length < listB.length ? listA : listB
-    const largestList = listA.length > listB.length ? listA : listB
-
-    return smallestList.reduce((participants, participant) => {
-      const isParticipantInLargeList = largestList.includes(participant)
-
-      if (isParticipantInLargeList) participants.push(participant)
+  intersectParticipants: (groupedParticipants) => {
+    const sortedGroupParticipants = groupedParticipants.sort(
+      (groupA, groupB) =>
+        groupA.length < groupB.length || groupB.length < groupA.length ? -1 : 0
```

Comment:
this logic is confusing to me. Is this saying that as long as groupA is not equal to groupB the value is -1, otherwise it is 0? That what it seems like with `groupA.length < groupB.length || groupB.length < groupA.length`.

Reply:
What this is doing is sorting the groups in ascending order by length. We want the smallest list in the front, before starting to intersect the participants. 

The reason for the -1 is because the sort function can take a predicate, and it should return 1, -1, or 0. These values are the directive by which to sort the elements in the array. In this case -1 would mean ascending order by length.

Reply:
I understand how the sort method works, what is confusing to me is the specific logic in the function. I think we can make it a lot clearer. If the array lengths are equal return 0, if array B is larger than array A return 1, else return -1. Or something along those lines.

Reply:
Thanks for that, i'll make the changes

Reply:
[0600b42](https://github.com/AMP-SCZ/dpdash/pull/756/commits/0600b42b64e6d112237f789162367af986d9a3db)